addons/web/static/tests/core/utils/html.test.js

```diff
@@ -40,6 +42,46 @@ test("createElementWithContent keeps html markup", () => {
     expect(res.outerHTML).toBe("<div><p>test</p></div>");
 });
 
+test("getInnerHtml escapes text and attributes", () => {
+    const div = document.createElement("div");
+    div.textContent = 'class="test" <i>abc</i>';
+    const span = document.createElement("span");
+    span.setAttribute("title", "<b>test</b>");
+    span.setAttribute("data-test", '" hack="failed');
+    div.appendChild(span);
+    const res = getInnerHtml(div);
+    expect(res.toString()).toBe(
+        'class=&quot;test&quot; &lt;i&gt;abc&lt;/i&gt;<span title="&lt;b&gt;test&lt;/b&gt;" data-test="&quot; hack=&quot;failed"></span>'
+    );
```

Comment:
For reference, here is what would be the output if using `innerHTML` directly rather than the new function:

<img width="1340" height="92" alt="image" src="https://github.com/user-attachments/assets/74f8dbbc-d11f-4036-8945-2d1cfc64a054" />